source/manager.go

```diff
@@ -32,7 +32,7 @@ type SourceInstance interface {
 	// CacheKey returns the cache key for the instance.
 	CacheKey(ctx context.Context, g session.Group, index int) (key, pin string, opts solver.CacheOpts, done bool, err error)
 
-	// Snapshot creates a cache ref for the instance.
+	// Snapshot creates a cache ref for the instance. May return a nil ref if there is no source content.
```

Comment:
"May return a nil ref if source points to empty content, e.g. image without any layers" maybe better

Reply:
@tonistiigi good idea!  Added in [77c3f85](https://github.com/moby/buildkit/pull/4202/commits/77c3f8572729914a1ffcee366f1afe8200c26124)

Reply:
Squash as well

Reply:
> Squash as well

nice; a good git history is really important!